### ISSUE TITLE
docs/typescript.md is out of date—instructions unclear

### ISSUE THREAD

**Issue by kjleitz** [rails/webpacker#2103]
**tl;dr: Do I still have to add the `appendTsSuffixTo: [/\.vue$/]` option?**

Currently the TypeScript doc is out of date. [It says the following](https://github.com/rails/webpacker/blame/master/docs/typescript.md#L25-L50):

> Change generated config/webpack/loaders/typescript.js from
>
>```js
>module.exports = {
>  test: /\.(ts|tsx)?(\.erb)?$/,
>  use: [{
>    loader: 'ts-loader'
>  }]
>}
>```
>
>to
>
>```js
>module.exports = {
>  test: /\.(ts|tsx)?(\.erb)?$/,
>  use: [{
>    loader: 'ts-loader',
>    options: {
>      appendTsSuffixTo: [/\.vue$/]
>    }
>  }]
>}
>```
>
>and now you can use <script lang="ts"> in your .vue component files.

But the generated output is actually:

```js
const PnpWebpackPlugin = require('pnp-webpack-plugin')

module.exports = {
  test: /\.(ts|tsx)?(\.erb)?$/,
  use: [
    {
      loader: 'ts-loader',
      options: PnpWebpackPlugin.tsLoaderOptions()
    }
  ]
}
```

I'm not sure exactly what [`PnpWebpackPlugin`](https://github.com/arcanis/pnp-webpack-plugin) does even after reading the README.md—seems that you can add extra options to `ts-loader` by passing them as a parameter to `tsLoaderOptions` but I don't know if it's already taking care of the `appendTsSuffixTo: [/\.vue$/]` option...? Documentation for `pnp-webpack-plugin` is sparse.

_Side note:_ `test: /\.(ts|tsx)?(\.erb)?$/,` seems wrong. It'll match `foo.ts.erb`, but it'll also match `foo..erb` and `foo.`. That should probably be `test: /\.(ts|tsx)(\.erb)?$/,`!

**Comment by jakeNiemiec** [comment]
> I'm not sure exactly what PnpWebpackPlugin does even after reading the README.md

See https://github.com/yarnpkg/rfcs/pull/101

**Comment by connorshea** [comment]
You can put the options Object in `PnpWebpackPlugin.tsLoaderOptions()`, like so:

```javascript
const PnpWebpackPlugin = require('pnp-webpack-plugin')

module.exports = {
  test: /\.(ts|tsx)?(\.erb)?$/,
  use: [
    {
      loader: 'ts-loader',
      options: PnpWebpackPlugin.tsLoaderOptions({
        appendTsSuffixTo: [/\.vue$/]
      })
    }
  ]
}
```

That's probably what the docs should be updated to say.
